### PR TITLE
crimson: implement privilege dropping

### PR DESCRIPTION
**Pull request by rzarzynski**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@rzarzynski sorry for the latency. just got a chance to review your changeset. it looks good to me. could you rebase it on main? and then it would be good for a rados qa-run.

**Comment by tchaikov** [comment]
@rzarzynski why did you mark this change with "DNM"?

File: src/global/privileges_drop.cc [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
+#ifdef WITH_SEASTAR
```
Comment (tchaikov) [inline src/global/privileges_drop.cc:4]: @rzarzynski just a nit, might want to fix the commit message:

> thatbrakes

**Review by tchaikov — APPROVED** [review]

**Comment by rzarzynski** [comment]
jenkins retest this please

```
[PostBuildScript] - [INFO] Executing post build scripts.
[PostBuildScript] - [INFO] Build does not have any of the results [ABORTED]. Did not execute build step #0.
INFO: Processing CTest-Version 3.x (default)
INFO: [CTest-Version 3.x (default)] - No test report file(s) were found with the pattern 'build/Testing/**/Test.xml' relative to '/home/jenkins-build/build/workspace/ceph-pull-requests' for the testing framework 'CTest-Version 3.x (default)'.
Did you enter a pattern relative to (and within) the workspace directory?
Did you generate the result report(s) for 'CTest-Version 3.x (default)'?"
WARNING: All test reports are empty.
INFO: Check 'Failed Tests' threshold.
INFO: Setting the build status to SUCCESS
[Checks API] No suitable checks publisher found.
Setting status of 29761dfda17b338f9df72bf25279b6cce51338e1 to FAILURE with url https://jenkins.ceph.com/job/ceph-pull-requests/99883/ and message: 'make check failed
```

File: src/global/privileges_drop.cc (hunk 1 above)
Comment (rzarzynski) [reply src/global/privileges_drop.cc:4]: Fixed!

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test api

File: src/global/privileges_drop.cc [hunk 2]
```diff
@@ -7,27 +7,47 @@
...
     if (g_conf()->setuser.length()) {
-      cerr << "ignoring --setuser " << g_conf()->setuser << " since I am not root"
-	   << std::endl;
+      logger.error("ignoring --setuser {} since I am not root",
+		   g_conf()->setuser);
     }
     if (g_conf()->setgroup.length()) {
-      cerr << "ignoring --setgroup " << g_conf()->setgroup
-	   << " since I am not root" << std::endl;
+      logger.debug("ignoring --setgroup {} since I am not root",
```
Comment (liu-chunmei) [inline src/global/privileges_drop.cc:52]: should be logger.error here?

**Pull request closed without merging**